src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -476,29 +476,61 @@ private void AddCentralTransitiveDependencyGroupsForPackageReference(PackageSpec
 
             // Do not pack anything from the runtime graphs
             // The runtime graphs are added in addition to the graphs without a runtime
-            foreach (var targetGraph in targetGraphs.Where(targetGraph => string.IsNullOrEmpty(targetGraph.RuntimeIdentifier)))
+            foreach (RestoreTargetGraph targetGraph in targetGraphs.Where(targetGraph => string.IsNullOrEmpty(targetGraph.RuntimeIdentifier)))
             {
-                var centralPackageVersionsForFramework = project.TargetFrameworks.Where(tfmi => tfmi.FrameworkName.Equals(targetGraph.Framework)).FirstOrDefault()?.CentralPackageVersions;
+                TargetFrameworkInformation targetFrameworkInformation = project.TargetFrameworks.FirstOrDefault(i => i.FrameworkName.Equals(targetGraph.Framework));
+
+                if (targetFrameworkInformation == null)
+                {
+                    continue;
+                }
 
                 // The transitive dependencies enforced by the central package version management file are written to the assets to be used by the pack task.
                 IEnumerable<LibraryDependency> centralEnforcedTransitiveDependencies = targetGraph
-                    .Flattened
-                    .Where(graphItem => graphItem.IsCentralTransitive && centralPackageVersionsForFramework?.ContainsKey(graphItem.Key.Name) == true)
-                    .Select((graphItem) =>
+                    .Graphs
+                    .SelectMany(i => i.InnerNodes)
+                    .Where(node => node?.Item != null && node.Disposition == Disposition.Accepted && node.Item.IsCentralTransitive && targetFrameworkInformation.CentralPackageVersions?.ContainsKey(node.Item.Key.Name) == true)
+                    .Select((node) =>
```

Comment:
Can this lambda function be a static method, to ease debugging in the future?

```suggestion
                    .Select(node => CreateLibraryDependency(node))
```

Reply:
I personally prefer static methods because those encapsulate logic in a separate place. 

I think lambda as long as it does not modify any of the variables created outside of the lambda function. I just want to avoid any unwanted side effects.